src/site/content/en/blog/serial/index.md

```diff
@@ -253,6 +253,94 @@ while (true) {
 }
 ```
 
+To specify the buffer into which data is read instead of the stream allocating a buffer for each chunk, get a [ReadableStreamBYOBReader] by calling `port.readable.getReader({ mode: "byob" })` and read data into a new `ArrayBuffer`. Note that the Web Serial API supports "Bring Your Own Buffer" readers in Chrome 106 or later.
+
+```js
+try {
+  const reader = port.readable.getReader({ mode: "byob" });
+  let buffer = new ArrayBuffer(512);
+  // Call reader.read() to read data into the buffer...
+} catch (error) {
+  if (error instanceof TypeError) {
+    // BYOB readers are not supported.
+    // Fallback to port.readable.getReader()...
+  }
+}
+```
+
+Here's an example of how to reuse the buffer out of `value.buffer`:
+
+```js
```

Comment:
This example ends up being kinda complicated because it's combining a bunch of things together. It handles transient errors with the `while (port.readable)` and also has a fallback if BYOB buffers aren't supported. Do you think we should pull it all together into a single example or keep it short and just demonstrate one concept. Like,

```js
const bufferSize = 8 * 1024; // 8kB
let buffer = new ArrayBuffer(bufferSize);

const reader = port.readable.getReader({ mode: 'byob' });
while (true) {
  const { value, done } = await reader.read(new Uint8Array(buffer));
  if (done) {
    break;
  }
  buffer = value.buffer;
  // Handle `value`.
}
reader.releaseLock();
```

Reply:
I've kept it short. WDYT?